### PR TITLE
Frontend Authentication/Sign In

### PR DESCRIPTION
**Pull request by NathanielB123**
Plus some infrastructure  for supporting links to specific pages

### CODE REVIEW

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -1,103 +1,105 @@
...
 import { Navbar } from "./Components/Navbar";
 import { ColumnGrid } from "./Components/ColumnGrid";
 import { WorkStationGroup } from "./Data";
+import React, { useState } from "react";
+import { BrowserRouter, Navigate, Route, Routes } from "react-router-dom";
+import { MainView } from "./Pages/MainView";
+import { STATS_PATH } from "./Config/Paths";
 
-const API_URL = "";
+export const API_URL = "http://localhost:8000";
```
Comment (danielg0) [inline frontend/src/App.tsx]: I know this is still draft but this change will break deployment. I think we either need a way to use Docker for dev, or some way to make this configurable

Comment (NathanielB123) [reply frontend/src/App.tsx]: This is a good point. I have no idea what the best way to handle this would be though. @aDotInTheVoid @Euphoride any ideas?

Comment (Euphoride) [reply frontend/src/App.tsx]: 100% should be an environment variable. Unsure whether the current docker configuration uses this to provide the API URL but this is the easiest way that comes to mind for dev/prod.

Comment (aDotInTheVoid) [reply frontend/src/App.tsx]: yeah, it’ll depend on your build system, but their should be a pay to pull out env vars at build time

**Comment by NathanielB123** [comment]
(Merge commit because Jacob has branched off this branch and I am NOT dealing with having to rebase both branches)

**Comment by aDotInTheVoid** [comment]
woo! now let’s never do this ever again

**Pull request merged**